singleheader/amalgamate.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python3
```

Comment:
`/usr/bin/env python` is recommended.

Reply:
Please enlighten me! Why, and why python instead of python3?

Reply:
Python is very rarely actually at `/usr/bin/python3`. `/usr/bin/env` is reliably present, and if I had for example installed Python with Homebrew, this would work unlike `/usr/bin/python3`. If I was in a Python `virtualenv` and ran this as ./amalgamate.py instead of `python amalgamate.py`, it would also use a Python other than the one the user would have intended.

`python3` is fine. On systems where python3 is installed and selected as the default, `python` is an alias to `python3`. Your tool does indeed depend on Py3.5+ as you're using pathlib.

Reply:
So `/usr/bin/env python3`  then?

Reply:
yep, fixed and pushed.